app/attributes/attributecontroller.cpp

```diff
@@ -442,6 +465,26 @@ bool AttributeController::isNewFeature() const
   return FID_IS_NULL( mFeatureLayerPair.feature().id() );
 }
 
+void AttributeController::acquireId()
+{
+  if ( !mFeatureLayerPair.layer() )
+    return;
+
+  startEditing();
+  QgsFeature feat = mFeatureLayerPair.feature();
+  if ( !mFeatureLayerPair.layer()->addFeature( feat ) )
+  {
+    QgsMessageLog::logMessage( tr( "acquireId: feature could not be added" ),
+                               QStringLiteral( "Input" ),
+                               Qgis::Critical );
+
+  }
+
+  connect( mFeatureLayerPair.layer(), &QgsVectorLayer::featureAdded, this, &AttributeController::onFeatureAdded );
+  commit();
```

Comment:
maybe it is properly handled already, just checking - what happens if commit() fails? Do we show any error message?

The `mFeatureLayerPair.layer()->addFeature( feat )` call generally should not be failing, only when something is really wrong in the app logic, but commit() may fail because provider could reject something (for example if one of the fields has a constraint (e.g. "unique") unfulfilled. If not handled well, let's defer to a separate ticket, just pointing it out so we don't forget about it, as this may be silently loosing data...

Reply:
Hm, yes, it is **not** being handled yet. How do you think we should notify user? Toast message or the popup Viktor made for GPS accuracy?

Reply:
I think the correct approach would be to show a big error popup that saving failed, and ideally also detailed message from `layer->commitErrors()`. Also, we should not close the form on failed save - the user may know how to fix the problem and try saving again...

Visually I am thinking about something like this (just a random image from internet, does not need to look like that)

![image](https://user-images.githubusercontent.com/193367/124123261-62e3ed00-da77-11eb-9faf-bb2fec597e6f.png)

Reply:
Added the check to https://github.com/lutraconsulting/input/issues/1530